cmd/bootstrap/run/qc.go

```diff
@@ -59,34 +66,44 @@ func GenerateRootQC(block *flow.Block, votes []*model.Vote, participantData *Par
 		createdQC = qc
 	})
 	if err != nil {
-		return nil, fmt.Errorf("could not CombinedVoteProcessor processor: %w", err)
+		return nil, nil, fmt.Errorf("could not CombinedVoteProcessor processor: %w", err)
 	}
 
+	invalidVotes := make([]error, 0, len(votes))
 	// STEP 2: feed the votes into the vote processor to create QC
 	for _, vote := range votes {
 		err := processor.Process(vote)
+
+		// in case there are invalid votes, we continue process more votes,
+		// so that finalizing block won't be interrupted by any invalid vote.
+		// if no enough votes are collected, finalize will fail and exit anyway, because
+		// no QC will be built.
 		if err != nil {
-			return nil, fmt.Errorf("fail to process vote %v for block %v from signer %v: %w",
+			if model.IsInvalidVoteError(err) {
+				invalidVotes = append(invalidVotes, err)
+				continue
+			}
+			return nil, invalidVotes, fmt.Errorf("fail to process vote %v for block %v from signer %v: %w",
 				vote.ID(),
-				block.ID(),
+				vote.BlockID,
 				vote.SignerID,
 				err)
 		}
 	}
 
 	if createdQC == nil {
-		return nil, fmt.Errorf("QC is not created, total number of votes %v, expect to have 2/3 votes of %v participants",
+		return nil, invalidVotes, fmt.Errorf("QC is not created, total number of votes %v, expect to have 2/3 votes of %v participants",
 			len(votes), len(identities))
 	}
```

Comment:
If we decide to log the invalid vote errors rather than returning `[]error`, we could shift this logic to separate vote processing function to add a unit test for it.